server/src/main/java/com/lunchtime/service/impl/PersonServiceImpl.java

```diff
@@ -1,24 +1,53 @@
 package com.lunchtime.service.impl;
 
+import com.lunchtime.mapper.PersonMapper;
 import com.lunchtime.models.Person;
+import com.lunchtime.service.dto.PersonDto;
+import com.lunchtime.service.dto.PersonPassDto;
 import com.lunchtime.repository.PersonRepository;
 import com.lunchtime.service.PersonService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 
 @Service
+@RequiredArgsConstructor
 public class PersonServiceImpl implements PersonService {
     private final PersonRepository personRepository;
-
-    public PersonServiceImpl(PersonRepository personRepository) {
-        this.personRepository = personRepository;
-    }
+    private final PersonMapper personMapper;
 
     public Person savePerson(Person person) {
         return personRepository.save(person);
     }
 
+    public PersonDto updatePerson(PersonDto personDto) {
+        Person person = personMapper.fromDtoToPerson(personDto);
+        personRepository.save(person);
+        return personDto;
+    }
+
+    public PersonPassDto updatePassword(PersonPassDto personPassDto) throws Exception {
+        Optional<Person> result = personRepository.findById(personPassDto.getId());
+        if (result.isPresent()) {
+            Person person = result.get();
+            if (personPassDto.getOldPassword().equals(person.getPassword())) {
+                person.setPassword(personPassDto.getPassword());
+                person.setName(personPassDto.getName());
+                person.setPhoneNumber(personPassDto.getPhoneNumber());
+                personRepository.save(person);
+                return personPassDto;
```

Comment:
I think PersonService shouldn't have a logic for mapping. It just has to work with repository. 
Something like this:
```
        Person oldPerson = personRepository.findPersonById(personPasswordDto.getId());
        if (personPasswordDto.getOldPassword().equals(oldPerson.getPassword())) {
            Person newPerson = personPasswordMapper.fromPasswordDtoToPerson(personPasswordDto, oldPerson);
            personRepository.save(newPerson);
            return personPassDto;
        }
```
But we need also PersonPasswordMapper here. Yes, and write not `pass` but `password` so that there won't be any misunderstanding when another developer may look your code.

Reply:
Also, if you can deal with this bug, it will be great. When I type new username, my avatar color is switching to another colors with each input symbol.